Packs/CyberArk/Integrations/CyberArk/CyberArk.py

```diff
@@ -0,0 +1,934 @@
+    demisto.log(f"There are {total_users} users")
+    results = CommandResults(
+        outputs_prefix='CyberArk.Users',
+        outputs_key_field='id',
+        outputs=response.get("Users")
+    )
+    return_results(results)
+
+def activate_user_command(
+        client: Client,
+        userID: str,
+):
+    response = client.activate_user(userID)
+    return response, {}, []
+
+def suspend_user_command(
+        client: Client,
+        userID: str,
+):
+    response = client.suspend_user(userID)
+    return response, {}, []
+
+
+def list_safes_command(
+        client: Client,
+):
+    response = client.list_safes()
+    total_safes = response.get("Total")
+    demisto.log(f"There are {total_safes} safes")
+    results = CommandResults(
+        outputs_prefix='CyberArk.Safes',
+        outputs_key_field='SafeUrlId',
+        outputs=response.get("Safes")
+    )
+    return_results(results)
+
+
+def get_safe_by_name_command(
+        client: Client,
+        safe_name: str,
+):
+    response = client.get_safe_by_name(safe_name)
+    results = CommandResults(
+        outputs_prefix=f'CyberArk.{safe_name}',
+        outputs_key_field='',
+        outputs=response
+    )
+    return_results(results)
+
+
+def add_safe_command(
+        client: Client,
+        safe_name: str,
+        description: str = "",
+        OLAC_enabled: str = "true",
+        managing_cmp: str = "",
+        number_of_versions_retention: str = "",
+        number_of_days_retention: str = "",
+        location: str = ""
+):
+    response = client.add_safe(safe_name, description, OLAC_enabled, managing_cmp, number_of_versions_retention,
+                               number_of_days_retention, location)
+    results = CommandResults(
+        outputs_prefix=f'CyberArk.{safe_name}',
+        outputs_key_field='',
+        outputs=response
+    )
+    return_results(results)
+
+def update_safe_command(
+        client: Client,
+        safe_name: str,
+        safe_new_name: str = "",
+        description: str = "",
+        OLAC_enabled: str = "true",
+        managing_cmp: str = "",
+        number_of_versions_retention: str = "",
+        number_of_days_retention: str = "",
+        location: str = ""
+):
+    response = client.update_safe(safe_name, safe_new_name, description, OLAC_enabled, managing_cmp, number_of_versions_retention,
+                                  number_of_days_retention, location)
+    results = CommandResults(
+        outputs_prefix=f'CyberArk.{safe_name}',
+        outputs_key_field='',
+        outputs=response
+    )
+    return_results(results)
+
+
+def delete_safe_command(
+        client: Client,
+        safe_name: str,
+):
+    response = client.delete_safe(safe_name)
+    # the response should be an empty string, if an error raised it would be catch in the main block
+    # should never enter to the else block, extra precautions if something want wrong
+    if not response:
+        return_results(f"Safe {safe_name} was deleted")
```

Comment:
shouldn't we add a `deleted: True` key to the context in this case?

Reply:
wasn't sure about it... sounds legit I will add it ( to here and to all the relevant delete commands.